sklearn/isotonic.py

```diff
@@ -252,7 +278,7 @@ def _build_f(self, X, y):
             # single y, constant prediction
             self.f_ = lambda x: y.repeat(x.shape)
         else:
-            self.f_ = interpolate.interp1d(X, y, kind='slinear',
+            self.f_ = interpolate.interp1d(X, y, kind='linear',
```

Comment:
The change to `slinear`  was introduced in #4111 to deal with duplicate minimal values.

However we since had "_make_unique" introduced by @amueller that might have been able to fix that.

I believe the non regression test for the issue fixed in #4111 is supposed to be `test_isotonic_regression_ties_min` but reading the code it seems to me that
it is not testing what it's supposed to test. I suggest changing it to:

```diff
diff --git a/sklearn/tests/test_isotonic.py b/sklearn/tests/test_isotonic.py
index 5058ecd..c5c45ec 100644
--- a/sklearn/tests/test_isotonic.py
+++ b/sklearn/tests/test_isotonic.py
@@ -98,9 +98,9 @@ def test_isotonic_regression():

 def test_isotonic_regression_ties_min():
     # Setup examples with ties on minimum
-    x = [0, 1, 1, 2, 3, 4, 5]
-    y = [0, 1, 2, 3, 4, 5, 6]
-    y_true = [0, 1.5, 1.5, 3, 4, 5, 6]
+    x = [1, 1, 2, 3, 4, 5]
+    y = [1, 2, 3, 4, 5, 6]
+    y_true = [1.5, 1.5, 3, 4, 5, 6]

     # Check that we get identical results for fit/transform and fit_transform
     ir = IsotonicRegression()
```

I checked on my box and this test still pass with the new code in this PR (with recent scipy), travis will run the tests with old scipy just in case.

Also the fact that other non-regression tests such as `test_isotonic_regression_ties_secondary_` that compare with the results of R and that the calibration curve examples yield the same output makes me rather confident that this change will not reintroduce old bugs.